### PR TITLE
Set systemd mode if entrypoint begins with /bin/sh -c

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/13324

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by Luap99** [comment]
Looks like the conclusion in this issue was that we should not rely on this behaviour.
cc @vrothberg @giuseppe

**Comment by rhatdan** [comment]
Well I disagree with them for this specific case.

**Comment by vrothberg** [comment]
> Unfortunately this is not guaranteed by the POSIX standard. Most shells do that, but we can't really rely on that.

Relying on "random" behavior of some shells is certainly dangerous.  If you really really want this, please add a warning to the man pages so a) we don't forget about it and b) users are responsible if it doesn't work.

File: pkg/specgen/generate/container_create.go [hunk 1]
```diff
@@ -304,7 +304,16 @@ func createContainerOptions(ctx context.Context, rt *libpod.Runtime, s *specgen.
 				"/usr/sbin/init":       true,
 				"/usr/local/sbin/init": true,
 			}
-			if useSystemdCommands[command[0]] || (filepath.Base(command[0]) == "systemd") {
+			// Grab last command incase this is launched from a shell
+			cmd := command
+			if len(command) > 2 {
+				// Podman build will add "/bin/sh" "-c" to
```
Comment (vrothberg) [inline pkg/specgen/generate/container_create.go:310]: Somehow related: It's probably something `podman build` shouldn't do

**Comment by rhatdan** [comment]
I am not sure how random this is, I think you would be hard pressed to find a single image in docker.io that did not work correctly where `/bin/sh -c "foobar"` was not just and exec of foobar as PID1. I am sure you can come up with some shell where this does not happen, but it is not likely there is an image with a symlink to /bin/sh, and less likely that it would be running systemd as the PID 2 of the container.

File: pkg/specgen/generate/container_create.go (hunk 1 above)
Comment (rhatdan) [reply pkg/specgen/generate/container_create.go:310]: This is the standard set by Docker.

**Comment by rhatdan** [comment]
@giuseppe @flouthoc @Luap99 @vrothberg @mheon @containers/podman-maintainers PTAL

**Review by flouthoc — APPROVED** [review]
Patch LGTM but I think it would not work for shells which hardcode something else instead of [`/bin/sh`, `-c`, ...] in entrypoint.

Should we warn if final array `command` still contains any systemd command and `if useSystemd == false`

so maybe something like this in the end after processing everything or could be something in troubleshooting docs.
```go
if !useSystemd  {
   for _, cmd := range command {
       if useSystemCommands[cmd] {
           log.warn("Entrypoint might be expecting systemd but container is not configured to do so
           );
        }
    }
 }
 ```
 
 Otherthan this nit LGTM but its a non-blocker.

**Comment by flouthoc** [comment]
@giuseppe @vrothberg PTAL

**Review by giuseppe — APPROVED** [review]
/lgtm

**Pull request merged**